bin/main.rb

```diff
@@ -1,27 +1,88 @@
-puts "Welcome to Tic Tac Toe!"
-puts "-----------------------"
-
-puts "Play a game with (1) Computer, (2) Your Friend:"
-game_mode = gets.chomp
-
-puts "You selected #{game_mode}"
-
-puts "Select (X) or (O)"
-x_or_o = gets.chomp
-
-if (x_or_o != 'x' || x_or_o !='X' || x_or_o != 'o' || x_or_o != 'O')
-    puts "Please select (X) or (O)"
-    x_or_o = gets.chomp
-end
-
-# Main Game Loop, for now we countdown from 10 to 0 to test user interactivity
-# We will ask the user to enter a grid co ordinate between 1 and 9 and display the 
-# board/game state and request player 2 to enter the same and continue until the end of
-# the game (which happens when a player wins or all the cells are filled and no one has won).
-
-i = 10
-
-i.times do
-    puts "Enter Co-ordinate: "
-    input = gets.chomp
-end
\ No newline at end of file
+#! /usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative '../lib/board.rb'
+require_relative '../lib/game.rb'
+require_relative '../lib/player.rb'
+
+def color(text,color="default")
+    colors = {"default"=>"38","red"=>"31","green"=>"32","blue"=>"34"}
+    color_code = colors[color]
+    return "\033[#{color_code}m#{text}\033[0m"
+end
+
+def welcome
+    puts " #{color("Welcome to the tic-tac-toe game!.
+    1. The game is played on a grid that's 3 squares by 3 squares.
+    2. Player 1 is X and Player 2 is O.
+    3. Players take turns putting their marks in the numbered squares.
+    4. The first player to get 3 in a row wins.
+    5. When all 9 squares are full, the game is a draw. If no player has 3 in a row.","green")} "
+end
+welcome
+
+def display_board(n)
+    array = n.each_slice(3).to_a
+    puts ''
+    puts "#{color("---|---|---","blue")}"
+    array.each do |i|
+        line = ''
+        i.each_with_index do |j,k|
+            line += " #{j} #{'|'}" if k!=2
+            line += " #{j}" if k==2
+        end
+        puts "#{color("#{line}","blue")}"
+        puts "#{color("---|---|---","blue")}"
+    end
+    puts ''
+end
+
+loop do
+    game = Game.new
+
+    players = []
+
+    puts ''
+    2.times do |i|
+        print "#{color("Enter your name player #{i+1}: ","blue")}"
+        players << gets.chomp
+    end
+
+    game.player1.name = players[0]
+    game.player2.name = players[1]
+

```

Comment:
That line `41`
```ruby
game = Game.new
```
Should come after the line `53`

Why ? Because you can initialize the Game with the players as parameters as array.